RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromL1TTracksEDProducer.cc

```diff
@@ -0,0 +1,215 @@
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTrack.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/RectangularPixelTopology.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+
+#include "RecoTracker/CkfPattern/interface/BaseCkfTrajectoryBuilderFactory.h"
+#include "RecoTracker/CkfPattern/interface/BaseCkfTrajectoryBuilder.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTrackerEvent.h"
+#include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
+#include "RecoTracker/TkTrackingRegions/interface/GlobalTrackingRegion.h"
+
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/GeomPropagators/interface/StateOnTrackerBound.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/MeasurementDet/interface/MeasurementDet.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <vector>
+
+class dso_hidden SeedGeneratorFromL1TTracksEDProducer : public edm::stream::EDProducer<> {
+public:
+  SeedGeneratorFromL1TTracksEDProducer(const edm::ParameterSet& cfg);
+  ~SeedGeneratorFromL1TTracksEDProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void findSeedsOnLayer(const GeometricSearchDet& layer,
+                        const TrajectoryStateOnSurface& tsosAtIP,
+                        Propagator& propagatorAlong,
+                        const TTTrack<Ref_Phase2TrackerDigi_>& l1,
+                        edm::ESHandle<Chi2MeasurementEstimatorBase>& estimatorH,
+                        unsigned int& numSeedsMade,
+                        std::unique_ptr<std::vector<TrajectorySeed>>& out) const;
+
+  void produce(edm::Event& ev, const edm::EventSetup& es) override;
```

Comment:
```suggestion
  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
```
the signature changes when changing to edm::global module type